nautobot/extras/jobs.py

```diff
@@ -240,10 +239,13 @@ def properties_dict(cls):
 
     @classmethod
     def _get_vars(cls):
-        vars = OrderedDict()
-        for name, attr in cls.__dict__.items():
-            if name not in vars and issubclass(attr.__class__, ScriptVariable):
-                vars[name] = attr
+        vars = dict()
+        base_classes = reversed(inspect.getmro(cls))
+        attr_names = [name for base in base_classes for name in base.__dict__.keys()]
+        for name in attr_names:
+            attr_class = getattr(cls, name, None).__class__
+            if name not in vars and issubclass(attr_class, ScriptVariable):
+                vars[name] = getattr(cls, name)
```

Comment:
Since this is a bit magical and it took some discussion to arrive at this solution, can you please provide some commenting on why this is necessary and what is actually happening here?

Reply:
Added a docstring and a comment

```py
    @classmethod
    def _get_vars(cls):
        """
        Return dictionary of ScriptVariable attributes defined on this class and any base classes to the top of the inheritance chain.
        The variables are sorted in the order that they were defined, with variables defined on base classes appearing before subclass variables.
        """
        vars = dict()
        # get list of base classes, including cls, in reverse method resolution order: [BaseJob, Job, cls]
        base_classes = reversed(inspect.getmro(cls))
        attr_names = [name for base in base_classes for name in base.__dict__.keys()]
        for name in attr_names:
            attr_class = getattr(cls, name, None).__class__
            if name not in vars and issubclass(attr_class, ScriptVariable):
                vars[name] = getattr(cls, name)

        return vars
```